framework/framework.go

```diff
@@ -15,24 +15,34 @@ import (
 // Config represents the configuration used to create a new operator framework.
 type Config struct {
 	// Dependencies.
-	Logger    micrologger.Logger
-	Resources []Resource
+
+	// Initializer is to prepare the given context for a single reconciliation
+	// loop. Operators can implement common context packages to enable
+	// communication between resources. These context packages can be set up
+	// within the initializer. The initializer receives the custom object being
+	// reconciled. Information provided by the custom object can be used to
+	// initialize the context.
+	Initializer func(ctx context.Context, obj interface{}) (context.Context, error)
```

Comment:
Can we call it something like `InitReconcileContextFunc`? I'd like express that this is only about context initialization, and it is ran per reconcile loop in the name.

I also think this is a setting and could have default noop value.

Reply:
I am not sure we can find a good name to express all this because it is getting super unsexy. I thought the godoc would be good enough to clear things. What about `ContextInitializer`?

Reply:
I'm a big fan of short names, but anyway like `ContextInitializer` better. I like `Func` suffixes though. They're used everywhere in Go. So context `InitContextFunc` or `ContextInitFunc` would be my preference.

Up to you. It's your PR. Approved already.

Reply:
`InitCtxFunc` it is now.